messaging/src/main/java/org/axonframework/tracing/NestingSpanFactory.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.common.BuilderUtils;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.Message;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+/**
+ * Implementation of {@link SpanFactory} that wraps another factory and creates nested spans when handling messages
+ * instead of starting a new trace.
+ * <p>
+ * This factory includes a time limit (by default 2 minutes, but is configurable on the builder) that handling an event
+ * will become part of the dispatching trace. After this time limit it is considered its own trace. The time limit
+ * prevents replays of events from being added to the original trace, even after a longer period of time.
+ * The time limit only applies for events and does not affect commands and queries. These will always be part of the
+ * same trace.
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.6.3
+ */
+public class NestingSpanFactory implements SpanFactory {
```

Comment:
As it doesn't "nest everything", shouldn't we specify this as an `EventNestingSpanFactory`, for example?

Reply:
It does nest everything. Except for old event messages. 

So deadlines: Yes
So commands: Yes
So events: Yes... unless > 2 minutes old

Reply:
Ah great, thanks for clarifying that. I'll resolve this comment then.